controllers/appsv1/deployment_webhook.go

```diff
@@ -37,7 +37,7 @@ func NewDeploymentInterceptorWebhook(c client.Client) webhook.AdmissionHandler {
 }
 
 // You need to ensure the path here match the path in the marker.
-// +kubebuilder:webhook:path=/mutate-v1-deployment,mutating=true,failurePolicy=fail,groups="apps",resources=deployments,sideEffects=None,verbs=create;update,versions=v1,name=deployment.sidecar-injector.jaegertracing.io,admissionReviewVersions=v1
+// +kubebuilder:webhook:path=/mutate-v1-deployment,mutating=true,failurePolicy=ignore,groups="apps",resources=deployments,sideEffects=None,verbs=create;update,versions=v1,name=deployment.sidecar-injector.jaegertracing.io,admissionReviewVersions=v1
```

Comment:
+1.

Do we still need the label exclusion conf in the webhook if we will be using `ignore` `failurePolicy`? 

```yaml
webhooks:
  - name: deployment.sidecar-injector.jaegertracing.io
    objectSelector: # Skip resources with the name jaeger-operator
      matchExpressions:
        - key: app.kubernetes.io/name
          operator: NotIn
          values:
            - "jaeger-operator"
```

Reply:
really need probably not. But doesn't it make sense to at least exclude the operator? [Isito excludes](https://github.com/istio/istio/blob/098cc47dfcbdca202208bd241d9dee35cd230a4d/manifests/charts/istio-control/istio-discovery/templates/mutatingwebhook.yaml) even more to avoid to many (in this case pods) pass their webhooks.

And should we then keep `app.kubernetes.io/name`?

What do you think?

Reply:
No objection, perhaps then we can remove `app.kubernetes.io/name` and use something more generic. Is istio using a generic label/annotation to bypass the injection?

Reply:
Everything without a label to inject or a label injected doesnt pass the webhook.